cache-tree.c

```diff
@@ -791,13 +792,18 @@ static void prime_cache_tree_rec(struct repository *r,
 			cnt++;
 		else {
 			struct cache_tree_sub *sub;
+			struct strbuf subtree_path = STRBUF_INIT;
 			struct tree *subtree = lookup_tree(r, &entry.oid);
+
 			if (!subtree->object.parsed)
 				parse_tree(subtree);
 			sub = cache_tree_sub(it, entry.path);
 			sub->cache_tree = cache_tree();
-			prime_cache_tree_rec(r, sub->cache_tree, subtree);
+			strbuf_addf(&subtree_path, "%s%s/", tree_path->buf, entry.path);
```

Comment:
`strbuf_addf()` should generally be avoided for performance reasons. This is a really fast path, so I was only seeing ~8ms in `prime_cache_tree()` even for a full monorepo without sparse index, but let's use our best tricks when possible.

See derrickstolee/git@bf2a9d3 for the fixup.

Reply:
Pulled and squashed the fixup (and removed the extra `struct strbuf subtree_path = STRBUF_INIT;`).